scipy/stats/tests/test_stats.py

```diff
@@ -6785,131 +6785,159 @@ def test_obrientransform():
     assert_array_almost_equal(result[0], expected, decimal=4)
 
 
-def check_equal_gmean(array_like, desired, *, xp, axis=None, dtype=None, rtol=1e-7,
-                      weights=None):
+def check_equal_xmean(*args, xp, mean_fun, axis=None, dtype=None,
+                      rtol=1e-7, weights=None):
     # Note this doesn't test when axis is not specified
     dtype = dtype or xp.float64
+    if len(args) == 2:
+        array_like, desired = args
+    else:
+        array_like, p, desired = args
     array_like = xp.asarray(array_like, dtype=dtype)
     desired = xp.asarray(desired, dtype=dtype)
     weights = xp.asarray(weights, dtype=dtype) if weights is not None else weights
-    x = stats.gmean(array_like, axis=axis, dtype=dtype, weights=weights)
+    args = (array_like,) if len(args) == 2 else (array_like, p)
+    x = mean_fun(*args, axis=axis, dtype=dtype, weights=weights)
     xp_assert_close(x, desired, rtol=rtol)
 
 
-def check_equal_hmean(array_like, desired, axis=None, dtype=None, rtol=1e-7,
-                      weights=None):
-    x = stats.hmean(array_like, axis=axis, dtype=dtype, weights=weights)
-    assert_allclose(x, desired, rtol=rtol)
-    assert_equal(x.dtype, dtype)
+def check_equal_gmean(*args, **kwargs):
+    return check_equal_xmean(*args, mean_fun=stats.gmean, **kwargs)
 
 
-def check_equal_pmean(array_like, exp, desired, axis=None, dtype=None,
-                      rtol=1e-7, weights=None):
-    x = stats.pmean(array_like, exp, axis=axis, dtype=dtype, weights=weights)
-    assert_allclose(x, desired, rtol=rtol)
-    assert_equal(x.dtype, dtype)
+def check_equal_hmean(*args, **kwargs):
+    return check_equal_xmean(*args, mean_fun=stats.hmean, **kwargs)
+
 
+def check_equal_pmean(*args, **kwargs):
+    return check_equal_xmean(*args, mean_fun=stats.pmean, **kwargs)
 
+
+@array_api_compatible
 class TestHMean:
-    def test_0(self):
+    def test_0(self, xp):
         a = [1, 0, 2]
         desired = 0
-        check_equal_hmean(a, desired)
+        check_equal_hmean(a, desired, xp=xp)
 
-    def test_1d_list(self):
+    def test_1d_list(self, xp):
         #  Test a 1d list
```

Comment:
This test is possibly redundant with `test_1d_array` as `check_equal_xmean` does 
```
 array_like = xp.asarray(array_like, dtype=dtype)
 desired = xp.asarray(desired, dtype=dtype)
 weights = xp.asarray(weights, dtype=dtype) if weights is not None else weights
```